### ISSUE TITLE
Regarding the issue of adding adaptive noise

### ISSUE THREAD

**Issue by JinmuE** [JeffffffFu/Awesome-Differential-Privacy-and-Meachine-Learning#13]
Hello author, recently I have been trying to reproduce your paper: "Adap DP-FL: Differentially Private Federated
Learning with Adaptive Noise". I have found your project. When I was running the fed_avg_with_ldp. py file for this project, the accuracy of the model remained between 9% and 12%. The network model I am using is MNIST_CNN, with a local client dropout rate of 20. Could you please teach me how to properly configure and train the model. thank.

![Image](https://github.com/user-attachments/assets/2d40bde8-e17c-489a-af86-80f3e8cfd6ea)

**Comment by JeffffffFu** [comment]
I think this is depend on your privacy budget. And LDP is hard to balance the privacy and utility. I think the result is normal.

**Comment by JinmuE** [comment]
Oh, okay. thank you. I want to know which parameter in the code is the privacy budget. Furthermore, I did not fully understand the pseudocode in your paper.In step 8, each local customer needs to calculate gradients. Isn't gradient descent necessary here? Or is step 8 to record the last gradient after local training. In addition, w^k_t used SGD optimization in step 15. Was it only used once during training until the last step?

![Image](https://github.com/user-attachments/assets/48f60e54-254e-4d76-9cdb-51be8fed55ef)
